kmip/services/server/crypto/engine.py

```diff
@@ -148,6 +156,71 @@ def create_asymmetric_key_pair(self, algorithm, length):
         engine_method = self._asymetric_key_algorithms.get(algorithm)
         return engine_method(length)
 
+    def mac(self, algorithm, key, data):
+        """
+        Generate message authentication code.
+
+        Args:
+            algorithm(CryptographicAlgorithm): An enumeration specifying the
+                algorithm for which the MAC operation will use.
+            key(bytes): secret key used in the MAC operation
+            data(bytes): The data to be MACed.
+
+        Returns:
+            bytes: The MACed data
+
+        Raises:
+            InvalidField: Raised when the algorithm is unsupported or the
+                length is incompatible with the algorithm.
+            CryptographicFailure: Raised when the key generation process
+                fails.
+
+        Example:
+            >>> engine = CryptographyEngine()
+            >>> mac_data = engine.mac(
+            ...     CryptographicAlgorithm.HMAC-SHA256, b'\x01\x02\x03\x04')
```

Comment:
This example also needs `data`; it's only providing the `algorithm` and `key` right now.